### PR TITLE
    csi: add nodeExpandSecret KEP details

### PR DESCRIPTION
**Pull request by humblec**
Ref#
KEP: https://github.com/kubernetes/enhancements/pull/3173/
Implementation: https://github.com/kubernetes/kubernetes/pull/105963
Blog: https://github.com/kubernetes/website/pull/33979

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
Cc @xing-yang @sftim

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ different Kubernetes components.
 | `CSIMigrationvSphere` | `false` | Beta | 1.19 | |
 | `CSIMigrationPortworx` | `false` | Alpha | 1.23 | |
 | `csiMigrationRBD` | `false` | Alpha | 1.23 | |
+| `CSINodeExpandSecret` | `false` | Alpha | 1.25 | |
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:89]: Later in this page, there's a list of feature gates and their descriptions. Please add a description there for `CSINodeExpandSecret`.

File: content/en/docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -1228,6 +1228,18 @@ persistent volume:
...
+  `NodeExpandVolume` call. This field is optional, and may be empty if no
+  secret is required. If the object contains more than one secret, all
+  secrets are passed.  When you have configured secret data for node-initiated
+  volume expansion, the kubelet passes that data via the `NodeExpandVolume()`
+  call to the CSI driver. In order to use the `nodeExpandSecretRef` field, your
+  cluster should be running Kubernetes version 1.25 or later and you must enable
+  the [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+  named `CSINodeExpandSecret` for each kube-apiserver and for the kubelet on every
+  node. You must also be using a CSI driver that supports or requires secret data during
+  node-initiated storage resize operations.
```
Comment (sftim) [inline content/en/docs/concepts/storage/volumes.md:1242]: I'd sort at least the following 3 fields alphabetically:
- `nodeExpandSecretRef`
- `nodePublishSecretRef`
- `nodeStageSecretRef`

Would that be OK?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (humblec) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:89]: sure, done.

File: content/en/docs/concepts/storage/volumes.md (hunk 2 above)
Comment (humblec) [reply content/en/docs/concepts/storage/volumes.md:1232]: Sure, done :+1:

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -761,6 +762,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
   from the Portworx in-tree plugin to Portworx CSI plugin.
   Requires Portworx CSI driver to be installed and configured in the cluster.
 - `CSINodeInfo`: Enable all logic related to the CSINodeInfo API object in `csi.storage.k8s.io`.
+- `CSINodeExpandSecret`: Enable SecretRef to be passed to CSI driver for NodeExpandVolume CSI operation.
```
[the remark this answers is not in the document]
Comment (humblec) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: @sftim so why not me :) ,, thanks for the suggestion ( as always) , its better wording.. accommodated the changes..

**Comment by humblec** [comment]
@sftim do we need to get this merged before any deadline ?

**Comment by sftim** [comment]
The deadline for _merge_ is the v1.25 release date. However, there is also a deadline for having PRs ready to review (which is this week, I think).

**Comment by tengqm** [comment]
I believe this is good to go. Lift the hold if that is true.
/approve
/hold

**Comment by krol3** [comment]
@tengqm LGTM too, any improve or we can remomve the hold?

**Pull request merged**